File: Algorithms/Sorting/Bubble Sort/BubbleSort.java
class BubbleSort {
    public static int[] sort(int[] data) {
        int n = data.length;
        for (int i = 0; i < n; i++) {
            for (int j = 0; j < n - 1; j++) {
                if (data[j] > data[j + 1]) {
                    int temp    = data[j];
                    data[j]     = data[j + 1];
                    data[j + 1] = temp;
                }
            }
        }
        return data;
    }
}

File: Algorithms/Sorting/Bubble Sort/BubbleSort.go
func BubbleSort(data []int) {
	n := len(data)
	for i := 1; i < n-1; i++ {
		for j := n - 1; j >= i; j-- {
			if data[j] < data[j-1] {
				data[j], data[j-1] = data[j-1], data[i]
			}
		}
	}
}